examples/reshare/src/dkg/actor.rs

```diff
@@ -140,551 +97,338 @@ where
-                        Ordering::Equal => {
-                            // Process any final messages from other dealers/players.
-                            manager.process_messages(epoch.get()).await;
-
-                            // At the midpoint of the epoch, construct the deal outcome for inclusion.
-                            manager.construct_deal_outcome(epoch.get()).await;
+                match m {
+                    Message::Act { response } => {
+                        let outcome = dealer_result.clone();
+                        if outcome.is_some() {
+                            info!("including reshare outcome in proposed block");
                         }
-                        Ordering::Greater => {
-                            // Process any incoming deal outcomes from dealing contributors.
-                            manager.process_block(epoch.get(), block).await;
+                        if response.send(outcome).is_err() {
+                            warn!("dkg actor could not send response to Act");
                         }
                     }
+                    Message::Finalized { block, response } => {
+                        let epoch = compute_epoch(BLOCKS_PER_EPOCH, block.height);
+                        let relative_height =
+                            relative_height_in_epoch(BLOCKS_PER_EPOCH, block.height);
+                        let mid_point = BLOCKS_PER_EPOCH / 2;
+
+                        // Inform the orchestrator of the epoch exit after first finalization
+                        if relative_height == 0 {
+                            if let Some(prev) = epoch.previous() {
+                                orchestrator.report(orchestrator::Message::Exit(prev)).await;
+                            }
+                        }
 
-                    // Attempt to transition epochs.
-                    if let Some(epoch) = epoch_transition {
-                        let (next_dealers, next_public, next_share, success) =
-                            match manager.finalize(epoch.get()).await {
-                                (
-                                    next_dealers,
-                                    RoundResult::Output(Output { public, share }),
-                                    success,
-                                ) => (next_dealers, Some(public), Some(share), success),
-                                (next_dealers, RoundResult::Polynomial(public), success) => {
-                                    (next_dealers, Some(public), None, success)
-                                }
-                                (next_dealers, RoundResult::None, success) => {
-                                    (next_dealers, None, None, success)
+                        if let Some(log) = block.log {
+                            if let Some((dealer, log)) = log.check(&round_info) {
+                                // If we see our dealing outcome in a finalized block,
```

Comment:
This doesn't prevent us from ever posting twice (can post before previous is finalized) but does help. Let's improve the comment.